### PR TITLE
Update definitions for bindings package

### PR DESCRIPTION
**Pull request by BoresXP**
Please fill in this template.

- [ x ] Use a meaningful title for the pull request. Include the name of the package modified.
- [ x ] Test the change in your own code. (Compile and run.)
- [ x ] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [ x ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ x ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ x ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [ x ] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/TooTallNate/node-bindings/blob/master/bindings.js
- [ x ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ x ] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ x ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/bindings/index.d.ts [hunk 1]
```diff
@@ -1,13 +1,65 @@
...
+// Type definitions for bindings 1.5
 // Project: https://github.com/TooTallNate/node-bindings
-// Definitions by: Daniel Perez Alvarez <https://github.com/unindented>
+// Definitions by: Daniel Perez Alvarez <https://github.com/unindented>, Boris Gulay <https://github.com/BoresXP>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
+/// <reference types="node" />
+
+export interface BindingOptions {
+	/**
```
Comment (danger-public) [inline types/bindings/index.d.ts:9]: - :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.

**Review by BoresXP — COMMENTED** [review]
So, what?

File: types/bindings/index.d.ts [hunk 2]
```diff
@@ -1,13 +1,65 @@
...
  * The main `bindings()` function loads the compiled bindings for a given module.
  * It uses V8's Error API to determine the parent filename that this function is
  * being invoked from, which is then used to find the root directory.
+ * @param mod Module name or loading configuration
+ * @returns loaded module or undefined
  */
-declare function bindings(mod: string): any;
-
-export = bindings;
+export default function bindings(mod: string | BindingOptions): any;
```
Comment (sandersn) [inline types/bindings/index.d.ts:65]: bindings doesn't export a `default` property, so you should revert that part of the PR.

**Review by sandersn — CHANGES_REQUESTED** [review]
Also need a review from @unindented on the content of BindingOptions -- I'm not qualified to review that part.

File: types/bindings/index.d.ts (hunk 2 above)
Comment (BoresXP) [reply types/bindings/index.d.ts]: Fixed in 0c9b988

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## intro.js ([<kbd>unpkg</kbd>](https://unpkg.com/browse/intro.js@latest/))
was missing the following properties:
1. introJs
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 0c9b9881b779306992493ed680285c136f006691
</p>

File: types/bindings/index.d.ts [hunk 3]
```diff
@@ -1,13 +1,65 @@
...
 /**
  * The main `bindings()` function loads the compiled bindings for a given module.
  * It uses V8's Error API to determine the parent filename that this function is
  * being invoked from, which is then used to find the root directory.
+ * @param mod Module name or loading configuration
+ * @returns loaded module or undefined
  */
-declare function bindings(mod: string): any;
-
-export = bindings;
```
Comment (ExE-Boss) [inline types/bindings/index.d.ts:13]: This change is incorrect:

- <https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html>
- <https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require>

**Review by ExE-Boss — CHANGES_REQUESTED** [review]

**Comment by BoresXP** [comment]
Someone pushed same changes ahead of me. Nice service, guys! FU

**Pull request closed without merging**